### PR TITLE
refs #23211 - make PuppetCA token provider more resilient

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

**Comment by timogoebel** [comment]
This fixes three issues I found with the current implementation:

* The helper script calls the wronmg URL. Bummer.
* The autosigner implementation for `#disable` needs to return `nil`.
* The `certificate` setting is `nil` by default. If we specify this explicitly smart-proxy treats it as a required parameter.

**Comment by timogoebel** [comment]
@ekohl: Can you review and merge?

File: extra/puppet_sign.rb [hunk 1]
```diff
@@ -32,8 +32,8 @@ def log(message)
 protocol         = PUPPETCA[:enabled] == true ? 'https' : PUPPETCA[:enabled]
 port             = protocol == 'https' ? SETTINGS[:https_port] : SETTINGS[:http_port]
 fqdn             = Socket.gethostbyname(Socket.gethostname).first
-# e.g. https://hostname.localdomain.com:8443/puppet/ca/autosign
-uri              = URI.parse("#{protocol}://#{fqdn}:#{port}/puppet/ca/autosign")
+# e.g. POST https://hostname.localdomain.com:8443/puppet/ca/verify
+uri              = URI.parse("#{protocol}://#{fqdn}:#{port}/puppet/ca/verify")
```
Comment (juliantodt) [inline extra/puppet_sign.rb]: This should be `/puppet/ca/validate`. See https://github.com/theforeman/smart-proxy/blob/develop/modules/puppetca/puppetca_api.rb#L40

**Review by juliantodt — CHANGES_REQUESTED** [review]
Nice catches! Thanks for the fixes. While you're at it, could you also change https://github.com/theforeman/smart-proxy/blob/develop/modules/puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner.rb#L10 to an Integer as this currently breaks the certificate generation?

**Comment by timogoebel** [comment]
@juliantodt: Fixed your comments. Can you ack now?

**Review by juliantodt — APPROVED** [review]

**Comment by timogoebel** [comment]
@lzap: Do you mind doing a quick merge? This fixes some issues we found when testing this in a production deployment.

File: extra/puppet_sign.rb [hunk 2]
```diff
@@ -26,14 +26,14 @@ def log(message)
...
+  log('PuppetCA smart-proxy module is not enabled!')
   exit 1
 end
 protocol         = PUPPETCA[:enabled] == true ? 'https' : PUPPETCA[:enabled]
 port             = protocol == 'https' ? SETTINGS[:https_port] : SETTINGS[:http_port]
 fqdn             = Socket.gethostbyname(Socket.gethostname).first
-# e.g. https://hostname.localdomain.com:8443/puppet/ca/autosign
-uri              = URI.parse("#{protocol}://#{fqdn}:#{port}/puppet/ca/autosign")
+# e.g. POST https://hostname.localdomain.com:8443/puppet/ca/validate
+uri              = URI.parse("#{protocol}://#{fqdn}:#{port}/puppet/ca/validate")
```
Comment (lzap) [inline extra/puppet_sign.rb:36]: This is quite a change I can hardly comment but this looks like some `extra` script.

**Review by lzap — APPROVED** [review]
I am good thanks.

**Comment by lzap** [comment]
There you have it! Thanks.

**Pull request merged**